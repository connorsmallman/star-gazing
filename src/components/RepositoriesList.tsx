import React from "react";
import {RepositoryListItem} from "./RepositoriesListItem";

export type Repository = {
  name: string;
  description: string;
  stars: number;
  id: number;
  url: string;
  isFavorite?: boolean;
}

type RepositoryListProps = {
  repositories: Repository[],
  favoriteRepository: (id: number) => void
  unfavoriteRepository: (id: number) => void
}

export const RepositoryList = ({ repositories, favoriteRepository, unfavoriteRepository }: RepositoryListProps) => {
  return (
    <>
      {repositories.map((repository: Repository) => {
        return (
          <RepositoryListItem
            key={repository.id}
            id={repository.id}
            stars={repository.stars}
            description={repository.description}
            name={repository.name}
            isFavorite={repository.isFavorite || false}
            handleOnClick={() => {
              window.location.href = repository.url;
            }}
            favoriteRepository={favoriteRepository}
            unfavoriteRepository={unfavoriteRepository}
          />
        );
      })}
  </>);
};