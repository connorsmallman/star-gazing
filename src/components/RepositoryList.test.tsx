import React  from 'react';
import {fireEvent, render, screen} from '@testing-library/react'

import {RepositoryList} from "./RepositoriesList";

const repositories = [
  {
    id: 1,
    name: 'InnoDB',
    isFavorite: false,
    description: 'Re-contextualized grid-enabled portal',
    stars: 345841,
    url: '',
  },
  {
    id: 2,
    name: 'BLACKHOLE',
    description: 'Inverse contextually-based portal',
    isFavorite: true,
    stars: 194518,
    url: '',
  },
  {
    id: 3,
    name: 'MyISAM',
    isFavorite: false,
    description: 'Fundamental multi-tasking standardization',
    stars: 19127,
    url: '',
  }
]

describe('Repository List', () => {
  test('displays list of repositories', () => {
    const list = render(<RepositoryList repositories={repositories} favoriteRepository={jest.fn()} unfavoriteRepository={jest.fn()} />);
    expect(list).toMatchSnapshot();
  });

  test('favorite repository', async () => {
    const favorite = jest.fn();
    render(<RepositoryList repositories={repositories} favoriteRepository={favorite} unfavoriteRepository={jest.fn()} />)

    fireEvent.click(screen.getAllByText('Favorite')[0]);

    expect(favorite).toBeCalledWith(repositories[0].id);
  });

  test('unfavorite repository', async () => {
    const unfavorite = jest.fn();
    render(<RepositoryList repositories={repositories} favoriteRepository={jest.fn()} unfavoriteRepository={unfavorite} />)

    fireEvent.click(screen.getAllByText('Unfavorite')[0]);

    expect(unfavorite).toBeCalledWith(repositories[1].id);
  });
});
