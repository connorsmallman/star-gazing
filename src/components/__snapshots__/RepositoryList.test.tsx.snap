// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository List displays list of repositories 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
      >
        <div
          class="flex flex-col"
        >
          <h2
            class="font-bold mb-2"
          >
            InnoDB
          </h2>
          <p
            class="mb-2"
          >
            Re-contextualized grid-enabled portal
          </p>
          <div>
            <svg
              class="inline-block align-middle fill-gray-800 mr-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
              />
            </svg>
            <span
              class="inline-block align-middle"
            >
              345841
            </span>
          </div>
          <div
            class="mt-4"
          >
            <button
              class="p-2 rounded-md bg-gray-100"
            >
              <svg
                class="inline-block"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
              <span
                class="inline-block align-middle ml-1"
              >
                Favorite
              </span>
            </button>
          </div>
        </div>
      </article>
      <article
        class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
      >
        <div
          class="flex flex-col"
        >
          <h2
            class="font-bold mb-2"
          >
            BLACKHOLE
          </h2>
          <p
            class="mb-2"
          >
            Inverse contextually-based portal
          </p>
          <div>
            <svg
              class="inline-block align-middle fill-gray-800 mr-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
              />
            </svg>
            <span
              class="inline-block align-middle"
            >
              194518
            </span>
          </div>
          <div
            class="mt-4"
          >
            <button
              class="p-2 rounded-md bg-gray-100"
            >
              <svg
                class="inline-block"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="inline-block align-middle ml-1"
              >
                Unfavorite
              </span>
            </button>
          </div>
        </div>
      </article>
      <article
        class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
      >
        <div
          class="flex flex-col"
        >
          <h2
            class="font-bold mb-2"
          >
            MyISAM
          </h2>
          <p
            class="mb-2"
          >
            Fundamental multi-tasking standardization
          </p>
          <div>
            <svg
              class="inline-block align-middle fill-gray-800 mr-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
              />
            </svg>
            <span
              class="inline-block align-middle"
            >
              19127
            </span>
          </div>
          <div
            class="mt-4"
          >
            <button
              class="p-2 rounded-md bg-gray-100"
            >
              <svg
                class="inline-block"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
              <span
                class="inline-block align-middle ml-1"
              >
                Favorite
              </span>
            </button>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
    >
      <div
        class="flex flex-col"
      >
        <h2
          class="font-bold mb-2"
        >
          InnoDB
        </h2>
        <p
          class="mb-2"
        >
          Re-contextualized grid-enabled portal
        </p>
        <div>
          <svg
            class="inline-block align-middle fill-gray-800 mr-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
          <span
            class="inline-block align-middle"
          >
            345841
          </span>
        </div>
        <div
          class="mt-4"
        >
          <button
            class="p-2 rounded-md bg-gray-100"
          >
            <svg
              class="inline-block"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
            <span
              class="inline-block align-middle ml-1"
            >
              Favorite
            </span>
          </button>
        </div>
      </div>
    </article>
    <article
      class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
    >
      <div
        class="flex flex-col"
      >
        <h2
          class="font-bold mb-2"
        >
          BLACKHOLE
        </h2>
        <p
          class="mb-2"
        >
          Inverse contextually-based portal
        </p>
        <div>
          <svg
            class="inline-block align-middle fill-gray-800 mr-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
          <span
            class="inline-block align-middle"
          >
            194518
          </span>
        </div>
        <div
          class="mt-4"
        >
          <button
            class="p-2 rounded-md bg-gray-100"
          >
            <svg
              class="inline-block"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
            <span
              class="inline-block align-middle ml-1"
            >
              Unfavorite
            </span>
          </button>
        </div>
      </div>
    </article>
    <article
      class="p-2 border rounded-sm mb-4 h-min-24 hover:cursor-pointer"
    >
      <div
        class="flex flex-col"
      >
        <h2
          class="font-bold mb-2"
        >
          MyISAM
        </h2>
        <p
          class="mb-2"
        >
          Fundamental multi-tasking standardization
        </p>
        <div>
          <svg
            class="inline-block align-middle fill-gray-800 mr-1"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
          <span
            class="inline-block align-middle"
          >
            19127
          </span>
        </div>
        <div
          class="mt-4"
        >
          <button
            class="p-2 rounded-md bg-gray-100"
          >
            <svg
              class="inline-block"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
            <span
              class="inline-block align-middle ml-1"
            >
              Favorite
            </span>
          </button>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
